app/presenters/specialist_document_presenter.rb

```diff
@@ -40,4 +61,23 @@ def protected_food_drink_name?
   def statutory_instrument?
     content_item.document_type == "statutory_instrument"
   end
+
+  def format_facet_value(type, value, key, filterable)
+    if type == "date"
+      view_context.display_date(value)
+    elsif type == "text" && value.is_a?(Array) && filterable == true
+      value.map { |v| format_filterable_links(key, v[:value], v[:label]) }
+    else
+      value
+    end
+  end
```

Comment:
I wonder if we could do this mapping in the view? (Just because this is the first time in the new consolidated routes that we've passed a view_context to a presenter rather than it just being available in views and helpers, so want to double-check whether this is a legit precedent to set).

Current precedent for it being in view: https://github.com/alphagov/frontend/pull/4225/files#diff-0b7fa2198812d14b54b691c4fde2f55cab0ba0109aaf95efaffba7e58e587d39R4-R10

Reply:
I went back and forth about this. I'm not happy about passing in a `view_context` but then I didn't want to duplicate the helper methods either. 

The reason I didn't put it directly in the view is because of the amount of logic in it, especially in the `elsif`. That `format_facet_value` method is called inside a loop and it would get messy and hard to debug.

I also felt it would be easier to test in a unit test rather than a system test.

Reply:
I'd be happy to pair on a better way to do this, as I'm sure there must be one.